### PR TITLE
Fix custom CA certificates for task/web/migration

### LINKED ISSUE #1782: AWX database migration job fail when using custom CA certificate
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx-operator/issues) for duplicates.
- [X] I understand that the AWX Operator is open source software provided for free and that I might not receive a timely response.

### Bug Summary

During the upgrade to awx 24.0.0, a new pod is created for migration. This pod doesnt contain the custom ca certificate and fails when using external postgres with verify enabled.

### AWX Operator version

2.13.1

### AWX version

24.0.0

### Kubernetes platform

kubernetes

### Kubernetes/Platform version

1.27.11

### Modifications

no

### Steps to reproduce

Update from awx 23.8.1 to 24.0.0
Add custom ca certs with external postgres. 

### Expected results

migration pod run with custom ca certs added and migration finishes in task pods.

### Actual results

no custom ca certs added and pod failing with django.db.utils.OperationalError: connection failed: certificate verify failed

### Additional information

  File "/var/lib/awx/venv/awx/lib64/python3.9/site-packages/psycopg/connection.py", line 728, in connect
    raise ex.with_traceback(None)
django.db.utils.OperationalError: connection failed: certificate verify failed

### Operator Logs

_No response_

### PR DESCRIPTION
**Pull request by JoelKle**
##### SUMMARY
This PR fixes the usage of custom CA certificates in the migration job and improves the existing implementation in the task & web deployment.
During the upgrade to awx 24.0.0, a new pod is created for migration. This pod doesnt contain the custom ca certificate and fails when using external postgres with verify enabled.

PR fixes: https://github.com/ansible/awx-operator/issues/1782
Follow up / Improves: https://github.com/ansible/awx-operator/pull/1800

##### ISSUE TYPE
 - Bug, Docs Fix or other nominal change

##### ADDITIONAL INFORMATION
This PR is based on the great work of @YassineFadhlaoui in https://github.com/ansible/awx-operator/issues/1782#issuecomment-2072044985 and @akkaba23 in https://github.com/ansible/awx-operator/pull/1800#issuecomment-2031485996

The following has changed:
- Added a new init container `init-bundle-ca-trust` to the `task` + `web` deployment and to the `migration` job
  - the init container runs the `update-ca-trust extract` if `bundle_ca_crt` is set.
  - I've added a new init container for the following reasons:
    - this init container need to run before all other init containers
    - only this init container requires `runAsUser: 0`

- Removed the `update-ca-trust` command from the init container `init-receptor` because it will run once in the new init container `init-bundle-ca-trust`

- Removed the mounting of the volume `{{ ansible_operator_meta.name }}-bundle-cacert` from containers that really do not need it

- Added the whole `bundle_ca_crt` logic to the `migration` job

I've successfully tested that change during my upgrade from awx-operator v2.12.1 to v2.15.0

### CODE REVIEW

**Comment by akkaba23** [comment]
Thank you for the nice work done on this PR @JoelKle .

**Comment by JoelKle** [comment]
@rooftopcellist @fosterseth Could you have a look on this PR? Thx :)

File: roles/installer/templates/deployments/web.yaml.j2 [hunk 1]
```diff
@@ -80,31 +80,43 @@ spec:
...
-{% if bundle_ca_crt or projects_persistence|bool or init_container_extra_commands %}
-        - name: init
+{% if bundle_ca_crt  %}
+        - name: init-bundle-ca-trust
           image: '{{ _init_container_image }}'
           imagePullPolicy: '{{ image_pull_policy }}'
           resources: {{ init_container_resource_requirements }}
+          securityContext:
+            runAsUser: 0
+            runAsGroup: 0
```
Comment (rooftopcellist) [inline roles/installer/templates/deployments/web.yaml.j2]: @JoelKle This will break on Openshift.  Can we template this in only if is_k8s=true?

File: roles/installer/templates/deployments/task.yaml.j2 [hunk 2]
```diff
@@ -74,6 +74,28 @@ spec:
...
 {% endif %}
       initContainers:
+{% if bundle_ca_crt  %}
+        - name: init-bundle-ca-trust
+          image: '{{ _init_container_image }}'
+          imagePullPolicy: '{{ image_pull_policy }}'
+          resources: {{ init_container_resource_requirements }}
+          securityContext:
+            runAsUser: 0
+            runAsGroup: 0
```
Comment (rooftopcellist) [inline roles/installer/templates/deployments/task.yaml.j2]: @JoelKle This will break on Openshift. Can we template this in only if is_k8s=true?

File: roles/installer/templates/deployments/web.yaml.j2 (hunk 1 above)
Comment (rooftopcellist) [reply roles/installer/templates/deployments/web.yaml.j2]: Did `update-ca-trust extract` require more permissions than just `update-ca-trust`?  I am trying to understand why the securityContext bit was added.

Comment (JoelKle) [reply roles/installer/templates/deployments/web.yaml.j2]: I removed the whole `securityContext` block from this PR. Now the init-container behaves as before.
@rooftopcellist Is this fine for you?

TL;DR
Withouth the `runAsUser: 0` parameter the init-container runs the update-ca-trust command as `uid=1000(awx) gid=0(root)`. That means the files in the folder `/etc/pki/ca-trust/extracted` are owned by `awx:root`
With the `runAsUser: 0` parameter all files are owned by `root:root` which is more accurate.

Comment (JoelKle) [reply roles/installer/templates/deployments/web.yaml.j2]: The `update-ca-trust extract` behaves the same as withouth the `extract`. However with the `extract` the command may print more warnings, ...
Check the manpage of update-ca-trust extract > "COMMANDS"

**Comment by JoelKle** [comment]
@rooftopcellist Are there any other blocker for you?

**Comment by rooftopcellist** [comment]
@JoelKle thanks for following up, and sorry for the delay here.  Could you please rebase to resolve the conflicts?  We'll see if CI passes on the run kicked off by pushing the rebased branch.  

I just checked the CI failure, and it looks like it was flake, so once rebased I think we will be good.

**Comment by JoelKle** [comment]
@rooftopcellist Rebase done. Look like the CI run needs your approval?

**Review by rooftopcellist — APPROVED** [review]
